dataprofiler/profilers/profile_builder.py

```diff
@@ -1442,14 +1442,16 @@ def _get_correlation(self, clean_samples):
         clean_column_ids = []
         if columns is None:
             for idx in range(len(self._profile)):
-                if (self._profile[idx].profile['data_type'] not in ['int', 'float']
-                        or self._profile[idx].null_count > 0):
+                data_type = self._profile[idx].\
+                    profiles["data_type_profile"].selected_data_type
+                if data_type not in ["int", "float"]:
                     clean_samples.pop(idx)
                 else:
                     clean_column_ids.append(idx)
 
-        data = pd.DataFrame(clean_samples)
-        data = data.apply(pd.to_numeric, errors='coerce')
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors='coerce')
+        means = {index:mean for index, mean in enumerate(batch_properties['mean'])}
+        data = data.fillna(value=means)
```

Comment:
does this apply at the same nan indices detected by our DP?

Reply:
Yes, since the means array in batch properties is lined up with the column indices

Reply:
I meant the row indices, not the column indices. As we added a lot of nan indices detection, I'm just wondering if those detected indices are consistent with the indices used with `fillna` ?

Reply:
cleaned_samples has those removed, hence the `na` here is for covering when the columns are merged and na fills in the empty locations